internal/clientsholder/clientsholder.go

```diff
@@ -64,6 +66,13 @@ func GetTestClientsHolder(k8sMockObjects []runtime.Object, filenames ...string)
 	return &clientsHolder
 }
 
+// GetTestClientsHolderAPIExt Overwrites the existing clientholders with a mocked version for unit testing.
+func GetTestClientsHolderAPIExt(k8sMockObjects []runtime.Object, filenames ...string) *ClientsHolder {
```

Comment:
The name is a bit misleading as what the function returns is the _clientsHolder_ itself, not the _APIExtClient_. If possible it may be better to unify the definition of all the fake clients for the test _clientsHolder_ in the funcion above (_GetTestClientsHolder()_) as it is done for the real one.

Reply:
What about passing a slice `mockObjects` with all the resources to the `GetTestClientsHolder()` function and then creating specific slices with the resources required/allowed for each client to be passed to them?

Reply:
I'll play around with the `reflect` library to see if there's something I can do there.

Reply:
Complete!  I used `reflect` to parse the incoming list of runtime objects and build individual slices for each client expecting them.